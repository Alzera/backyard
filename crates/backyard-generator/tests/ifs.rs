use backyard_generator::generate_serializable_node;
use backyard_parser::parse;

#[test]
fn basic() {
  let asts = parse(true, "if (false) {}").unwrap();
  insta::assert_yaml_snapshot!(generate_serializable_node(&asts).unwrap());
}

#[test]
fn with_else() {
  let asts = parse(true, "if (false) {} else {}").unwrap();
  insta::assert_yaml_snapshot!(generate_serializable_node(&asts).unwrap());
}

#[test]
fn with_elseif() {
  let asts = parse(true, "if (false) {} elseif (true) {}").unwrap();
  insta::assert_yaml_snapshot!(generate_serializable_node(&asts).unwrap());
}

#[test]
fn with_else_if() {
  let asts = parse(true, "if (false) {} else if (true) {}").unwrap();
  insta::assert_yaml_snapshot!(generate_serializable_node(&asts).unwrap());
}

#[test]
fn nested() {
  let asts = parse(true, "if (false) {if (false) {} else (true) {}} else if (true) {}").unwrap();
  insta::assert_yaml_snapshot!(generate_serializable_node(&asts).unwrap());
}

#[test]
fn no_block() {
  let asts = parse(true, "if (false) $e = 5; else if (true) $e = 6;").unwrap();
  insta::assert_yaml_snapshot!(generate_serializable_node(&asts).unwrap());
}

#[test]
fn short() {
  let asts = parse(true, "if (true): elseif (false): else: endif;").unwrap();
  insta::assert_yaml_snapshot!(generate_serializable_node(&asts).unwrap());
}
