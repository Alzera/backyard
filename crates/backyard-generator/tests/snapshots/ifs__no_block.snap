---
source: crates/backyard-generator/tests/ifs.rs
expression: generate(asts).unwrap()
snapshot_kind: text
---
"if (false) $e = 5;\nelseif (true) $e = 6;"
